docs/hashicorp_vault.md

```diff
@@ -0,0 +1,115 @@
+---
+id: hashicorp_vault
+title: HashicorpVault
+---
+
+# Hashicorp Vault
+
+The goal of this guide is to configure Signatory to use an Hashicorp Vault as a signing backend.
+
+## **Vault setup**
+
+Run Vault server or use dedicated cloud instance (HCP Vault) for secrets management
+
+### **Install vault**
+
+https://developer.hashicorp.com/vault/downloads
+
+### **Run Dev Server**
+
+```sh
+docker run --cap-add=IPC_LOCK -d --name=dev-vault vault
+```
+
+### **Configure Vault Address and Token**
+
+Set the Vault address and obtain the root token provided when starting the Vault server.
+
+```sh
+export VAULT_ADDR=http://127.0.0.1:8200
+export VAULT_TOKEN={{root_token}}
+```
+
+### **Enable Transit Secrets Engine**
+
+Enable the Transit secrets engine which is used to manage cryptographic functions.
+
+```sh
+vault secrets enable transit
+```
+
+### **Create Transit Key**
+
+Create a new encryption key in the transit secrets engine.
+
+```sh
+vault write -f transit/keys/{{my-key}} type="ed25519"
+```
+
+### **Create Policy for Transit**
+
+Create a policy that allows reading and writing keys in the transit engine.
+
+```hcl
+# transit-policy.hcl
+path "transit/*" {
+  capabilities = ["read", "create", "update", "list"]
+}
+```
+
+### **Create AppRole Authentication**
+
+Enable and configure the AppRole authentication method.
+
+```sh
+vault auth enable approle
+```
+
+### **Create AppRole Role**
+
+Create a role for the AppRole authentication method. This role associates a set of policies with the AppRole.
+
+```sh
+vault write auth/approle/role/my-approle \
+  secret_id_ttl=10m \
```

Comment:
`secret_id_ttl` needs to be zero for "does not expire"

Reply:
This document has a sample configuration for general understanding.
Such a fine configuration should be handled by the user himself, depending on the needs.
But here I agree, we can make it without expiration for test purposes

Reply:
I think it's important for the user to know that if/when `secret_id_ttl` expires, Signatory service does not start during next restart event